db/remote/db.go

```diff
@@ -48,10 +55,19 @@ func (d *DB) View(fn func(txn db.Transaction) error) error {
 }
 
 func (d *DB) Update(fn func(txn db.Transaction) error) error {
-	return db.Update(d, fn)
+	start := time.Now()
+
+	err := db.Update(d, fn)
+
+	if d.listener != nil {
+		d.listener.OnCommit(time.Since(start))
```

Comment:
after removing `d.listener` check consider using `defer func() { d.listener.OnCommit(...)} ` as in the rest of codebase (but only here)